React/todoApp/src/components/Item.jsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import styles from "./Item.module.css";
+
+function Item({ item, setTodos }) {
+  const [editing, setEditing] = React.useState(false);
+  const inputRef = React.useRef(null);
+
+  const toggleComplete = () => {
+    setTodos(prev =>
+      prev.map(t => (t.id === item.id ? { ...t, is_completed: !t.is_completed } : t))
+    );
+  };
+
+  const deleteTodo = () => {
+    setTodos(prev => prev.filter(t => t.id !== item.id));
+  };
+
+  const startEdit = () => setEditing(true);
+
+  React.useEffect(() => {
+    if (editing && inputRef.current) {
+      inputRef.current.focus();
+      const len = inputRef.current.value.length;
+      inputRef.current.setSelectionRange(len, len);
+    }
```

Comment:
Newlines between statements, returns and conditional blocks make the code more readable

```suggestion
    if (editing && inputRef.current) {
      inputRef.current.focus();
      
      const len = inputRef.current.value.length;
      
      inputRef.current.setSelectionRange(len, len);
    }
```